autoscheduler/user_sessions/views.py

```diff
@@ -4,6 +4,47 @@
 from rest_framework.parsers import JSONParser
 from django.contrib.auth.models import User
 
+def _set_state_in_session(request, key: str):
+    """ Function that sets the given key in our session to the value of the key in the
+        request body. Must have term and the given key in the body of the request.
+        Used for save_courses and save_availabilities
+    """
+    try:
+        body = json.loads(request.body.decode())
+        objs = body.get(key, {})
+        term = body.get('term')
+        if not term:
+            return Response(f'Request body must contain {key} and term', status=400)
+    except (UnicodeError, json.JSONDecodeError):
+        return Response(status=400)
+
+    # Attempt to get user's session
+    session = request.session
+
+    term_data = session.setdefault(term, {})
+    term_data[key] = objs
+    session.modified = True
+
+    return Response()
+
+def _get_state_from_session(request, key: str):
+    """ Retrieves the value for the respective key from the session. Must have a term as
+        a query parameter. Used for get_courses and get_availabilities.
+    """
+
+    term = request.query_params.get('term')
+    if not term:
+        return Response(status=400)
+
+    session = request.session
+    response = []
+    if term:
```

Comment:
The `if not term: return ...` acts as a guard statement above, so this is redundant + can be removed (noticed when working on #267 )